multiversx_sdk/facades/entrypoints.py

```diff
@@ -33,24 +28,38 @@
 from multiversx_sdk.wallet.user_verifer import UserVerifier
 
 
+class IAbi(Protocol):
+    def encode_endpoint_input_parameters(self, endpoint_name: str, values: List[Any]) -> List[bytes]:
+        ...
+
+    def encode_constructor_input_parameters(self, values: List[Any]) -> List[bytes]:
+        ...
+
+    def encode_upgrade_constructor_input_parameters(self, values: List[Any]) -> List[bytes]:
+        ...
+
+    def decode_endpoint_output_parameters(self, endpoint_name: str, encoded_values: List[bytes]) -> List[Any]:
+        ...
+
+
 class NetworkEntrypoint:
     def __init__(self,
-                 url: str,
-                 kind: str,
+                 network_provider_url: str,
+                 network_provider_kind: str,
                  chain_id: str) -> None:
-        if kind == "proxy":
-            self.provider = ProxyNetworkProvider(url)
-        elif kind == "api":
-            self.provider = ApiNetworkProvider(url)
+        if network_provider_kind == "proxy":
+            self.network_provider = ProxyNetworkProvider(network_provider_url)
+        elif network_provider_kind == "api":
+            self.network_provider = ApiNetworkProvider(network_provider_url)
         else:
             raise InvalidNetworkProviderKindError()
 
         self.chain_id = chain_id
-        self.factories_config = TransactionsFactoryConfig(self.chain_id)
 
-    def sign_transaction(self, transaction: Transaction, signer: UserSigner):
+    def sign_transaction(self, transaction: Transaction, account: UserSigner):
+        """Signs the transactions and applies the signature on the transaction."""
         tx_computer = TransactionComputer()
-        transaction.signature = signer.sign(tx_computer.compute_bytes_for_signing(transaction))
+        transaction.signature = account.sign(tx_computer.compute_bytes_for_signing(transaction))
```

Comment:
Type is still `UserSigner`, is this intended?

Reply:
changed to `Account`.